### DISCUSSION TITLE
[Bug] echo string with escaped newlines fails

### DISCUSSION THREAD

**Discussion by MrSkwiggs** [community/community#25469] [Actions]
<p>When running <code>echo ${{ toJson(github) }}</code> as a step, it fails when pushing a commit with multi-line commit message.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/6/9/69d8e8cf494372420870085ea6098f352f96c899.png" data-download-href="/uploads/short-url/f6mRmq0Wn6x932k12hMRSuUNWuR.png?dl=1" title="Screenshot 2020-10-09 at 09.30.31" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/6209874/181091248-a6f1eb9a-56c1-42c4-811d-13fb0b234ad8.png" alt="Screenshot 2020-10-09 at 09.30.31" data-base62-sha1="f6mRmq0Wn6x932k12hMRSuUNWuR" width="345" height="60" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/6/9/69d8e8cf494372420870085ea6098f352f96c899_2_345x60.png, https://aws1.discourse-cdn.com/github/optimized/3X/6/9/69d8e8cf494372420870085ea6098f352f96c899_2_517x90.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/6/9/69d8e8cf494372420870085ea6098f352f96c899_2_690x120.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/6/9/69d8e8cf494372420870085ea6098f352f96c899_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot 2020-10-09 at 09.30.31</span><span class="informations">900×158 12.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div><br>
<div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/b/e/be48860690982c3671f0e6ab0358b7c4917446a4.png" data-download-href="/uploads/short-url/r9k6dJMipbpIO4PqmPWKMPHR0wY.png?dl=1" title="Screenshot 2020-10-09 at 14.25.43" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/6209874/181091249-71dde233-8751-434d-b9ec-4eaba9be0da6.png" alt="Screenshot 2020-10-09 at 14.25.43" data-base62-sha1="r9k6dJMipbpIO4PqmPWKMPHR0wY" width="517" height="354" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/b/e/be48860690982c3671f0e6ab0358b7c4917446a4_2_517x354.png, https://aws1.discourse-cdn.com/github/optimized/3X/b/e/be48860690982c3671f0e6ab0358b7c4917446a4_2_775x531.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/b/e/be48860690982c3671f0e6ab0358b7c4917446a4_2_1034x708.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/b/e/be48860690982c3671f0e6ab0358b7c4917446a4_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot 2020-10-09 at 14.25.43</span><span class="informations">1700×1164 126 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div><br>
This is what immediately precedes the line where <code>timestamp</code> is <code>echo</code>ed
<pre><code class="lang-auto">"message": "Merge pull request #9 from MrSkwiggs/master\n\nUpdate test.yml",
"timestamp": "2020-10-09T09:36:42+02:00",
</code></pre>

**Comment by Simran-B** [comment, marked as answer]
<p>To avoid issues with multiple lines or command substitution, the documentation suggests to use <code>env</code>:</p>
<pre><code class="lang-auto">steps:
  - name: Print github context
    env:
      GITHUB_CONTEXT: ${{ toJSON(github) }}
    run: echo "$GITHUB_CONTEXT"
</code></pre>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions#example-printing-context-information-to-the-log-file" target="_blank" rel="noopener nofollow ugc">docs.github.com</a>
  </header>
  <article class="onebox-body">
    <img src="" class="thumbnail" width="16" height="16">

<h3><a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions#example-printing-context-information-to-the-log-file" target="_blank" rel="noopener nofollow ugc">Context and expression syntax for GitHub Actions - GitHub Docs</a></h3>



  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>An alternative in bash is to use cat with heredoc syntax and quotes around the delimiter definition to disable substitution:</p>
<pre><code class="lang-auto">steps:
  - name: Print github context
    run: |
      cat&lt;&lt;'EOF'
      ${{ toJSON(github) }}
      EOF
</code></pre>
